### PR TITLE
Fix closestTo should possibly return undefined

### LINKED ISSUE #1915: `closestTo` TS typings are incorrect
If you give the `closestTo` function an empty list of dates, it returns `undefined`. This is not represented in the TS typings.

### PR DESCRIPTION
**Pull request by V-Gutierrez**
Closes #1915 

closestTo can possibly return undefined and this is foreseen by typings

### CODE REVIEW

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Hey, thank you for the fix, but it won't have an effect because we generate typings from JSDoc definitions. Please update those instead.

**Comment by wenderjean** [comment]
@kossnocorp The test file hosted in [/src/closestTo/test.js](https://github.com/date-fns/date-fns/blob/master/src/closestTo/test.js) is still relevant? We're discussing about the assertions on lines `36`, `94` and `101` and they look pretty weird actually. Should we take the opportunity to work on them?

1. The assertion in the line `36` could refer itself to a `falsy` value considering the comparison
2. The assertions in the lines `94` and `101` seem to be false positives :thinking: 

What do you think?

**Comment by V-Gutierrez** [comment]
> @kossnocorp The test file hosted in [/src/closestTo/test.js](https://github.com/date-fns/date-fns/blob/master/src/closestTo/test.js) is still relevant? We're discussing about the assertions on lines `36`, `94` and `101` and they look pretty weird actually. Should we take the opportunity to work on them?
> 
> 1. The assertion in the line `36` could refer itself to a `falsy` value considering the comparison
> 2. The assertions in the lines `94` and `101` seem to be false positives thinking
> 
> What do you think?

That's right, if they still relevant I would gladly update these too !

**Comment by kossnocorp** [comment]
Yes, please fix the tests as well, much appreciated! 36 def should check for `=== undefined`. Regarding 94 and 101, I'm not sure what to do with it. Do you have any suggestions?

Thank you, guys!

**Comment by V-Gutierrez** [comment]
> Yes, please fix the tests as well, much appreciated! 36 def should check for `=== undefined`. Regarding 94 and 101, I'm not sure what to do with it. Do you have any suggestions?
> 
> Thank you, guys!

I choose to fix the tests for checking the current behavior of closestTo. Not all cases returns Invalid Date as said in docs but for this issue I believe those fixes will do for now.

File: src/closestTo/test.js [hunk 1]
```diff
@@ -91,18 +91,18 @@ describe('closestTo', function() {
     assert.deepEqual(result, new Date(2015, 7 /* Aug */, 31))
   })
 
-  it('converts undefined into empty array', function() {
+  it('returns undefined when undefined is passed as second argument', function() {
```
Comment (hjJunior) [inline src/closestTo/test.js:94]:
```suggestion
  it('returns undefined when the second argument is omitted', function() {
```

File: src/closestTo/test.js [hunk 2]
```diff
@@ -91,18 +91,18 @@ describe('closestTo', function() {
...
     var date = new Date(2014, 6 /* Jul */, 2).getTime()
     // $ExpectedMistake
     var result = closestTo(date, undefined)
-    assert(result == null)
+    assert(result === undefined)
   })
 
-  it('converts null into empty array', function() {
+  it('converts null into undefined', function() {
```
Comment (hjJunior) [inline src/closestTo/test.js:101]:
```suggestion
  it('returns undefined if the given array is null', function() {
```

**Comment by ekilah** [comment]
@V-Gutierrez thanks for picking up the issue I created! 🎉

**Comment by V-Gutierrez** [comment]
@kossnocorp Closing this one since I made further improvements in #1968

**Pull request closed without merging**